controllers/imagepolicy_controller.go

```diff
@@ -260,7 +262,7 @@ func (r *ImagePolicyReconciler) SetupWithManager(mgr ctrl.Manager, opts ImagePol
 	}
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&imagev1.ImagePolicy{}).
+		For(&imagev1.ImagePolicy{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
This is just an observation and is out of scope, but I'm wondering if we should support the reconcile request annotation for ImagePolicy in which case we'd also need to add the `ReconcileRequestedPredicate` in here. I think ImagePolicy is the only CR we don't support reconciliation requests for at the moment.